internal/proxy/oauthproxy.go

```diff
@@ -212,9 +214,13 @@ func NewRewriteReverseProxy(route *RewriteRoute, config *UpstreamConfig) *httput
 		}
 		director := httputil.NewSingleHostReverseProxy(target).Director
 
-		req.Header.Add("X-Forwarded-Host", req.Host)
+		if req.Header.Get("X-Forwarded-Host") == "" {
```

Comment:
Here is one more instance where a new X-Forwarded-Host is not added if there is already one in the request.